src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.cs

```diff
@@ -113,7 +113,27 @@ internal TarHeader(TarEntryFormat format, TarEntryType typeFlag, TarHeader other
         internal void InitializeExtendedAttributesWithExisting(IEnumerable<KeyValuePair<string, string>> existing)
         {
             Debug.Assert(_ea == null);
-            _ea = new Dictionary<string, string>(existing);
+            Debug.Assert(existing != null);
+
+            using IEnumerator<KeyValuePair<string, string>> enumerator = existing.GetEnumerator();
+            while (enumerator.MoveNext())
+            {
+                KeyValuePair<string, string> kvp = enumerator.Current;
+
+                int index = kvp.Key.IndexOfAny(new char[] { '=', '\n' });
+                if (index >= 0)
+                {
+                    throw new ArgumentException(SR.Format(SR.TarExtAttrDisallowedKeyChar, kvp.Key, kvp.Key[index] == '\n' ? "\\n" : kvp.Key[index]));
+                }
+                if (kvp.Value.IndexOf('\n') != -1)
```

Comment:
readability:

```suggestion
                if (kvp.Value.Contains('\n'))
```

Reply:
I agree it's more readable, but I am unsure if Contains is more or less performant compared to IndexOf. Let me find out.

Reply:
Contains is strictly better in this case.

Reply:
We actually have an analyzer for this:
https://github.com/dotnet/runtime/blob/9c02b25f08c8f1aaf7811446325ac9f873efa451/eng/CodeAnalysis.src.globalconfig#L622-L623
Do we know why it's not firing?

Reply:
No idea. @carlossanlop can you please check?

Reply:
The analyzer has a bug: it only runs when using `==` or `>=`. My code is using `!=`.

I opened an issue: https://github.com/dotnet/roslyn-analyzers/issues/6587